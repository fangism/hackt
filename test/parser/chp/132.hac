// "132.hac"
// accepts-invalid bug

defproc foo(int sel; chan?(bool) in[2]; chan!(bool) out) {
bool x;

chp {
x := true;
*[
[ x ->
	in?(x);	// should reject invalid!
	out!(x)
[] else  -> skip
]]
}
}

foo bar;
