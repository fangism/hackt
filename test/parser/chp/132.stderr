ERROR expecting a scalar channel reference, but got a 1-dim reference.  [11:2..3]
ERROR in CHP statement(s) at [11:2--12:8]
